java-lib/src/main/java/org/apache/flink/ml/jna/util/ReadWriteUtils.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.jna.util;
+
+import com.google.protobuf.GeneratedMessageV3;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.File;
+
+/** Utility methods for reading and writing stages. */
+public class ReadWriteUtils {
+    public static void saveMetadata(GeneratedMessageV3 protobufMessage, String path)
+            throws IOException {
+        File metadataFile = new File(path, "metadata");
+        if (!metadataFile.createNewFile()) {
+            throw new IOException("File " + metadataFile.toString() + " already exists.");
+        }
+
+        try(FileOutputStream output = new FileOutputStream(metadataFile)) {
+            protobufMessage.writeTo(output);
+        }
+    }
+
+    public static <T> void saveModelData(GeneratedMessageV3 protobufMessage, String path) throws IOException {
+        File metadataFile = new File(path, "modeldata");
```

Comment:
Could we change the directory structure of the model data files to be similar to Flink ML's approach?

Note that Flink ML assumes the model data files are located under the directory `$path/data/`. And there could be multiple files under this directory.

Reply:
OK. I'll make the change.